### ISSUE TITLE
Slim package size

### ISSUE THREAD

**Issue by mpharrigan** [mdtraj/mdtraj#982]
The testing/reference folder is 60 MB

```
[snip]
100K    ncinpcrd.rst7
104K    4OH9.pdb
112K    ala_dipeptide_trj
132K    frame0.binpos
152K    inpcrd
156K    frame0.binpos.h5
156K    frame0.nc
172K    1ncw.pdb.gz
172K    frame0.dcd
180K    alanine-dipeptide-explicit.pdb
192K    ala_dipeptide.pdb
196K    1am7_protein.pdb
212K    4K6Q.pdb
240K    GG-tip4pew.pdb
268K    mdcrd.nc
272K    g_sas_ref.dat
276K    frame0.lammpstrj
280K    monolayer.pdb
352K    alanine-dipeptide-explicit.prmtop
356K    freesolve_v0.3.tar.bz2
408K    3nch.pdb.gz
448K    frame0.trr
480K    1am7_corrected.xtc
480K    1am7_uncorrected.xtc
488K    well-mixed.hoomdxml
524K    frame0.mdcrd
552K    frame0.mdcrdbox
556K    frame0.gro
592K    frame0.xyz
628K    nitrogen.arc
692K    2EQQ.pdb
748K    frame0.pdb
820K    nitrogen.pdb
976K    tz2.truncoct.parm7
1.1M    tip3p_300K_1ATM.xtc
1.4M    monolayer.xtc
1.4M    overflow.crd
1.6M    test_hoomdxml.dcd
1.6M    water_hex.pdb.gz
1.8M    1vii_sustiva_water.pdb
1.9M    test_bad.nc
1.9M    test_good.nc
2.0M    frame0.dtr
2.6M    test.parm7
28M     3pqr_memb.psf
58M     total
```

@swails git blame is gitblaming you for 3pqr_memb.psf. Is there a smaller file that could be used for testing?

**Comment by swails** [comment]
I can make it bigger :)

**Comment by mpharrigan** [comment]
:100: MB

**Comment by swails** [comment]
Actually, if I use ParmEd, I think I can reach the target:

```Python
import parmed as pmd
(pmd.load_file('3pqr_memb.psf') * 4).save('mdtraj_target.psf')
```

`mdtraj_target.psf` should be just over the 100 MB mark :)

**Comment by swails** [comment]
Give me a second to find my serious hat and figure out why I added such a ginormous file in the first place.

**Comment by mpharrigan** [comment]
Take your time. I'm going to go ahead with #983 for v1.5.1 although it's inelegant

**Comment by swails** [comment]
This was added in #735, in response to #711.

One thing I'll point out is that this is a pure ASCII file.  As such, it is highly compressible (`gzip -9` drops the size to 2.0 MB, putting it second in terms of overall size behind *another* file I added -- I think I added the 4 largest files, although none are bigger than 2.6MB following compression).

So our options are:

1. Find (i.e., *make*) a smaller PSF file that has multiple CHARMM chains.
2. Compress the file and be satisfied with the 2.0 MB size. Git will have to store it in its entirety, but it should never change so we won't be bloating our history as long as it's left alone
3. Leave it as-is

This PSF file is actually a very good test for the PSF parser, and hits most of the corner cases you could possible encounter in the real world, all in the same file.  It has many chains (i.e., SEGIDs), and you have residues that differ *only* in SEGID (see CARC and CARD, which both have a single residue numbered 1 with the same name).  It has residue names that are absurdly long (BGLCNA), as well as ridiculously long atom type names (CC3162).

I believe someone reported this PSF file to me as one that ParmEd and/or MDTraj parsed incorrectly, so it got sucked into my test suite for my PSF parsers.  It would probably take me quite awhile to come up with a PSF file that is considerably smaller and still stresses the PSF parser to the same degree, so I'd like to find a way to keep it if possible.

I have a question about distribution -- are there *any* forms where we distributed an *uncompressed* archive of MDTraj files?  Because if the answer to that is 'no' (as I would hope it would be), then this file adds 2 MB to the total size (making it comparable to many other test files), and it only takes up space on users' hard drives where they unpack/install MDTraj (and 28 MB is really not that big of a file even on SSDs).

**Comment by mpharrigan** [comment]
Very good points. Compressed (tar.gz) filesize of the reference/ directory is 17 MB with psf and 15 MB without

**Comment by mpharrigan** [comment]
Files over 100k when compressed

```
100K    frame0.mdcrdbox.gz
100K    frame0.mdcrd.gz
136K    nitrogen.pdb.gz
148K    frame0.binpos.h5.gz
168K    2EQQ.pdb.gz
168K    test.parm7.gz
172K    1ncw.pdb.gz
176K    well-mixed.hoomdxml.gz
188K    nitrogen.arc.gz
196K    mdcrd.nc.gz
356K    1vii_sustiva_water.pdb.gz
356K    freesolve_v0.3.tar.bz2.gz
408K    3nch.pdb.gz
476K    1am7_corrected.xtc.gz
480K    1am7_uncorrected.xtc.gz
504K    overflow.crd.gz
1.1M    tip3p_300K_1ATM.xtc.gz
1.4M    monolayer.xtc.gz
1.5M    test_hoomdxml.dcd.gz
1.6M    water_hex.pdb.gz
1.8M    test_bad.nc.gz
1.8M    test_good.nc.gz
2.1M    3pqr_memb.psf.gz
```

**Comment by mpharrigan** [comment]
I think the goal of a smaller package is for ease of transmission rather than space on disk. Even compressed, mdtraj is < 1MB of code and 17 MB of testing data

**Comment by swails** [comment]
It's still a sizeable percentage of the release tarball (~11% of the size taken up by a single file).  Compressing the file *inside* the repository won't affect distribution size at all -- it will only drop the total size it takes up on users' hard drives (which makes me lean toward it not being worthwhile).

That file is still the biggest offender, but by a far narrower margin than it first appears.  Seems like death by a thousand cuts to me (and 17 MB is still pretty small).

**Comment by swails** [comment]
One thing I'll point out is that the unit tests, while useful for users to validate their installation, are far *more* useful to us while developing.  Users also can't just blindly run all tests and expect them to pass,  (do they all get skipped if they can't be run? they didn't used to be).

The other potential use for test files is for running interactive tutorials on the website (users can use one of the provided test files when trying out stuff like RMSD or PCA).  But if we think 17 MB is a little too big and also think that there will be a sizeable number of users that will have no use for the test files, we can always split out the tests to a separate, optional package (something like `mdtraj-test-data` or something).

**Comment by mpharrigan** [comment]
Making testing reference data "opt-in" is probably not a bad idea

**Comment by ajasja** [comment]
Hi!

```
So our options are:
1.    Find (i.e., make) a smaller PSF file that has multiple CHARMM chains.
2.    Compress the file and be satisfied with the 2.0 MB size. Git will have to store it in its entirety, but it should never change so we won't be bloating our history as long as it's left alone
3.    Leave it as-is
```
There are also some other options: 
4. Remove some or all of the water from the psf. (Get's the size below 18 Mb)
5. Move the testing data into a different repo/package (or remove it when creating the conda package).

Currently I have 5 copies of this file on my SSD (due to the way that conda works). 
So perhaps nr. 5 is worth thinking about, since most users will not run the test.

![psf-file](https://cloud.githubusercontent.com/assets/19926/14115486/7e068fc4-f5db-11e5-8403-723c9854b7fa.png)

Oh and thanks or developing MDTraj. It's really cool (especially when I think of all the TCL scripts that I won't have to write:)

**Comment by mpharrigan** [comment]
I like option 5 as well.

p.s. check out `conda clean` for removing unused packages

**Comment by ajasja** [comment]
Thanks; really useful 2GB of disk space reclaimed!:)

**Comment by mpharrigan** [comment]
The testing reference data is no longer included in the conda package as of 1.9.1. Clone this repo to run tests! New binary package size is 5 MB compressed
